pkg/cmd/upgrade/clustermanager/exec.go

```diff
@@ -0,0 +1,181 @@
+// Copyright Contributors to the Open Cluster Management project
+package clustermanager
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"open-cluster-management.io/clusteradm/pkg/cmd/upgrade/clustermanager/scenario"
+	"open-cluster-management.io/clusteradm/pkg/helpers"
+	"open-cluster-management.io/clusteradm/pkg/helpers/apply"
+	"open-cluster-management.io/clusteradm/pkg/helpers/printer"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/cmd/util"
+	version "open-cluster-management.io/clusteradm/pkg/helpers/version"
+)
+
+func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
+	klog.V(1).InfoS("init options:", "dry-run", o.ClusteradmFlags.DryRun, )
+	o.values = Values{
+			Registry:    o.registry,
+		}
+	
+
+	versionBundle, err := version.GetVersionBundle(o.bundleVersion)
+
+	if err != nil {
+		klog.Errorf("unable to retrive version ", err)
+		return err
+	}
+
+	o.values.BundleVersion = BundleVersion{
+		RegistrationImageVersion: versionBundle.Registration,
+		PlacementImageVersion:    versionBundle.Placement,
+		WorkImageVersion:         versionBundle.Work,
+		OperatorImageVersion:     versionBundle.Operator,
+	}
+
+	return nil
+}
+
+func (o *Options) validate() error {
+
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	apiExtensionsClient , err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	installed, err := helpers.IsClusterManagerInstalled(apiExtensionsClient)
+	if err != nil {
+		return err
+	}
+
+	if !installed {
+		return fmt.Errorf("clustermanager is not installed")
+	}
+	fmt.Printf("clustermanager installed. starting upgrade ")
+
+
```

Comment:
Why not allowing downgrade? that can be powerful

Reply:
downgrade may also involve rollback the elder version of CRD and rbac. This could be quite complicated since we need to consider store all different version of manifests somewhere...